### DISCUSSION TITLE
WebREPL connection in Thonny interfering with running program

### DISCUSSION THREAD

**Discussion by bonzi9** [thonny/thonny#3299] [General]
Since I am not sure if this is user error on my behalf I'll post this as a discussion first.

I just started experimenting with Thonny, running MicroPython on an ESP8266 board, great experience so far!
Having set up webREPL with the `webrepl_setup` command, it is working fine directly from the webpage. I can CTRL + C to abort the current program and afterwards I can issue `import machine` and `machine.reset()` to reset the ESP.
However in Thonny, setting it up to connect to WebREPL breaks the running program, even with the setting 'Interrupt working program on connect' turned off. As soon as Thonny is started or I issue the `machine.reset()` command it tries to (re)connect to webREPL in a way that interferes with the running program. My main.py is a basic test program to connect WiFi, serve a webpage with socket and turn an LED to a specific brightness by listening to HTTP GET requests. As soon as I run code from Thonny or issue a reset the LED does not change anymore from the GET request.

The main issue for me is that I want to remotely manage the ESP but I cannot get a working reboot because Thonny seems to intercept the webREPL in a different way than what the internal web page does. The only way I can reboot the ESP correctly is either;

- Closing Thonny immediately after issuing the `machine.reset()` command.
- Using the internal webREPL page and issue `machine.reset()`.
- Do a physical reset on the ESP.

It might as well just be interfering with the use of socket, I am fairly new to Python in general so I can't tell if this is expected behavior or not.

**Comment by aivarannamaa** [comment]
Thanks for this information! I'll try to reproduce and fix this later.
